sdk/apis/syncagent/v1alpha1/published_resource.go

```diff
@@ -176,6 +176,9 @@ type RelatedResourceSpec struct {
 	// Mutation configures optional transformation rules for the related resource.
 	// Status mutations are only performed when the related resource originates in kcp.
 	Mutation *ResourceMutationSpec `json:"mutation,omitempty"`
+
+	// Optional indicates whether the related resource must be referenced.
```

Comment:
I don't think that was the intention of the ticket. I thought the ticket was about whether the referenced object _exists_, not whether it was referenced?